Sources/Shared/NotificationRateLimitsAPI.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import PromiseKit
+
+public struct RateLimitResponse: Decodable {
+    public var target: String
+
+    public struct RateLimits: Decodable {
+        public var attempts: Int
+        public var successful: Int
+        public var errors: Int
+        public var total: Int
+        public var maximum: Int
+        public var remaining: Int
+        public var resetsAt: Date
+    }
+
+    public var rateLimits: RateLimits
+}
+
+public class NotificationRateLimitsAPI {
+    public class func rateLimits(pushID: String) -> Promise<RateLimitResponse> {
+        firstly { () -> Promise<URLRequest> in
+            do {
+                var urlRequest = URLRequest(url: URL(
```

Comment:
Fetching data from the web to show in Home Assistant should happen inside Home Assistant